### ISSUE TITLE
ASG "invalid" filter should ignore space after snapshot name

### ISSUE THREAD

**Issue by tim-elliott** [capitalone/cloud-custodian#947]
ASG was terminated because it was referencing an invalid snapshot name.  The LC had functioned successfully using the name in the format "snap-1234abcd " (note the space at the end), but it was flagged as invalid because that literal name didn't exist as a snapshot.

**Comment by scotwk** [comment]
@tim-elliott - can you share the policy/policies that caused this?  I'm trying to understand which field is the problem, and where the space got added in.  Thanks.

**Comment by kapilt** [comment]
@scotwk its the asg filter `invalid`

**Comment by tim-elliott** [comment]
@scotwk as Kapil said it is just using the "invalid" filter, as in:
```
  - name: asg-invalid-check
    resource: asg
    filters:
      - invalid
```
if the launch configuration has an extra space after the snapshot name like "snap-abcd1234 " instead of "snap-abcd1234", autoscaling will successfully launch instances and use the correct snapshot.  however, the "invalid" filter will match it and any actions will be applied.

**Comment by scotwk** [comment]
Sorry - I should have been more specific.  I understand what filter the request is being made for.  I didn't understand where the extra space got entered?  Was that done via the AWS web interface or was that a problem that cloud custodian created?

(And at the time I commented I wasn't sure which field was the problem, because I was able to make an ASG with whitespace at the end in various fields such as `LaunchConfigurationName`.  I am pretty sure that Tim was referring to the EBS Snapshot field, which is what I submitted a fix for).

PR is submitted (#991) for this one problem, although it feels like more work needs to be done because if any of the fields could have leading/trailing whitespace we need to safeguard for all of them.
